code/game/objects/items/melee/batons/classic.dm

```diff
@@ -0,0 +1,133 @@
+	force = 12 //9 hit crit
+	w_class = WEIGHT_CLASS_NORMAL
+	var/cooldown = 13
+	var/last_hit = 0
+	var/stun_stam_cost_coeff = 1.25
+	var/hardstun_ds = 1
+	var/softstun_ds = 0
+	var/stam_dmg = 30
+
+/obj/item/melee/classic_baton/attack(mob/living/target, mob/living/user)
+	if(!baton_stun(target, user))
+		return ..()
+
+// return true to disrupt normal melee attack chain
+/obj/item/melee/classic_baton/proc/baton_stun(mob/living/target, mob/living/user)
+	if(user.getStaminaLoss() >= STAMINA_SOFTCRIT)//CIT CHANGE - makes batons unusuable in stamina softcrit
+		to_chat(user, "<span class='warning'>You're too exhausted for that.</span>")//CIT CHANGE - ditto
+		return FALSE//CIT CHANGE - ditto
+
+	add_fingerprint(user)
+	if((HAS_TRAIT(user, TRAIT_CLUMSY)) && prob(50))
+		to_chat(user, "<span class ='danger'>You club yourself over the head.</span>")
+		user.Knockdown(60 * force)
+		if(ishuman(user))
+			var/mob/living/carbon/human/H = user
+			H.apply_damage(2*force, BRUTE, BODY_ZONE_HEAD)
+		else
+			user.take_bodypart_damage(2*force)
+		return TRUE
+	if(iscyborg(target))
+		return FALSE
+	if(!isliving(target))
+		return FALSE
+	if (user.a_intent == INTENT_HARM)
+		return FALSE
+	else
+		if(last_hit < world.time)
+			if(target.check_shields(src, 0, "[user]'s [name]", MELEE_ATTACK))
+				playsound(target, 'sound/weapons/genhit.ogg', 50, 1)
+				return TRUE
+			if(ishuman(target))
+				var/mob/living/carbon/human/H = target
+				if(check_martial_counter(H, user))
+					return TRUE
+			playsound(get_turf(src), 'sound/effects/woodhit.ogg', 75, 1, -1)
+			target.Knockdown(softstun_ds, TRUE, FALSE, hardstun_ds, stam_dmg)
+			log_combat(user, target, "stunned", src)
+			src.add_fingerprint(user)
+			target.visible_message("<span class ='danger'>[user] has knocked down [target] with [src]!</span>", \
+				"<span class ='userdanger'>[user] has knocked down [target] with [src]!</span>")
+			if(!iscarbon(user))
+				target.LAssailant = null
+			else
+				target.LAssailant = user
+			last_hit = world.time + cooldown
+			user.adjustStaminaLossBuffered(getweight())//CIT CHANGE - makes swinging batons cost stamina
+	return TRUE
+
+/obj/item/melee/classic_baton/telescopic
+	name = "telescopic baton"
+	desc = "A compact yet robust personal defense weapon. Can be concealed when folded."
+	icon = 'icons/obj/items_and_weapons.dmi'
+	icon_state = "telebaton_0"
+	lefthand_file = 'icons/mob/inhands/weapons/melee_lefthand.dmi'
+	righthand_file = 'icons/mob/inhands/weapons/melee_righthand.dmi'
+	item_state = null
+	slot_flags = ITEM_SLOT_BELT
+	w_class = WEIGHT_CLASS_SMALL
+	item_flags = NONE
+	force = 0
+	total_mass = TOTAL_MASS_NORMAL_ITEM
+
+/obj/item/melee/classic_baton/telescopic/ComponentInitialize()
+	. = ..()
+	AddComponent(/datum/component/transforming, ATTACK_SELF_TRANSFORM, .proc/on_transform)
+
+/obj/item/melee/classic_baton/baton_stun(mob/living/target, mob/living/user)
+	if(!(SEND_SIGNAL(src, COMSIG_IS_TRANSFORMED) & COMPONENT_IS_TRANSFORMED))
```

Comment:
Use `GetComponent` and check the `active` variable instead. The less snowflake signals, the better.